src/storage/storage.ts

```diff
@@ -134,6 +151,20 @@ export class Storage implements IStorage {
     return this.store.getAddressAtIndex(index);
   }
 
+  async getAddressPubkey(index: number): Promise<string> {
+    const addressInfo = await this.store.getAddressAtIndex(index);
+    if (addressInfo?.publicKey) {
+      // public key already cached on address info
+      return addressInfo.publicKey;
+    }
+
+    // derive public key from xpub
+    const accessData = await this._getValidAccessData();
+    const hdpubkey = new HDPublicKey(accessData.xpubkey);
+    const key: HDPublicKey = hdpubkey.deriveChild(index);
+    return key.publicKey.toString('hex');
+  }
```

Comment:
note: We only got the pubkey from the derived mainKey, this does not work for external signing where we only have the xpubkey.